ksqldb-functional-tests/src/test/java/io/confluent/ksql/test/driver/AssertExecutorMetaTest.java

```diff
@@ -0,0 +1,48 @@
+package io.confluent.ksql.test.driver;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import io.confluent.ksql.properties.with.CommonCreateConfigs;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.kafka.common.config.ConfigDef;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Most test coverage for the AssertExecutor is in the sql-tests/test.sql file
+ * which is run through the KsqlTesterTest tool. This class is here to ensure
+ * that the class evolves properly.
+ */
+public class AssertExecutorMetaTest {
+
+  /**
+   * These are excluded from coverage
+   */
+  private final Set<String> excluded = ImmutableSet.of(
+      CommonCreateConfigs.SOURCE_NUMBER_OF_PARTITIONS, // testing tool does not support partitions
+      CommonCreateConfigs.SOURCE_NUMBER_OF_REPLICAS    // testing tool does not support replicas
+  );
+
+  @Test
+  public void shouldCoverAllWithClauses() {
+    // Given:
+    final ConfigDef def = new ConfigDef();
+    CommonCreateConfigs.addToConfigDef(def, true, true);
+    final Set<String> allValidProps = def.names();
+
+    // When:
+    final Set<String> coverage = AssertExecutor.MUST_MATCH
+        .stream()
+        .map(sp -> sp.withClauseName)
+        .flatMap(Arrays::stream)
+        .collect(Collectors.toSet());
+
+    // Then:
+    assertThat(Sets.difference(allValidProps, coverage), Matchers.is(excluded));
```

Comment:
To get the full list you need to use `CreateConfigs.CONFIG_DEF` and `CreateAsConfigs.CONFIG_DEF`.  This is currently only testing configs that are common to both.

Reply:
I had thought about this, but I don't think it really makes sense - you can only assert sources for `CREATE AS` (it doesn't really make sense to assert for a `CREATE`) and you specify it using the same syntax as `CREATE` schemas, meaning that it won't actually compile if you use the `CREATE AS` configs (though at the moment there are no configs in `CREATE AS` that aren't in `CREATE`, and I'm not sure I understand why there would be)

Reply:
Hummm... I guess we can cross this bring if/when we come to it.